src/include/migraphx/matcher.hpp

```diff
@@ -124,14 +126,14 @@ template <class M>
 auto bind_match(M m, std::string name)
 {
     return make_function_matcher(
-        [=, name = std::move(name)](matcher_context& ctx,
-                                    instruction_ref ins) -> optional<instruction_ref> {
+        [=, name_ = std::move(name)](matcher_context& ctx,
```

Comment:
The compilation on Windows is failing because of the capture `name` is shadowing the function argument `name`. I need to provide different name for the captured variable.

Reply:
Why does shadowing cause the compilation to fail? I assume it would just be a warning.